Add support for querying IPMI power status

remixtj opened the pull request:
##### SUMMARY

User may want to query for power status of an ipmi device
instead of changing it. With ipmi_facts, current state of the
ipmi status is returned and user may, as example, loop until a
desired power status is reached.
We developed this option because we have some tasks to
perform on storage side on some machines that boot from SAN.
These operations require to have machines powered off.

##### ISSUE TYPE
- New Module pull request

##### COMPONENT NAME
ipmi_facts

##### ADDITIONAL INFORMATION
Check docs included in the module for a sample of usage of the new parameter

remixtj committed:
Add support for querying IPMI power status

User may want to query for power status of an ipmi device
instead of changing it. With 'query' state, current power
status is returned and user may, as example, loop until a
desired power status is reached.
We developed this option because we have some tasks to
perform on storage side on some machines that boot from SAN.
These operations require to have machines powered off.

jamescassell reviewed (changes requested):
'query' is not a 'state' -- this should be a separate module, likely _info or_facts

remixtj committed:
Reverted modules change

remixtj committed:
Added ipmi_facts module

Instead of using query option, as stated by previous
comments, i added a module for checking for ipmi status

remixtj committed:
Docs update

remixtj commented:
@jamescassell Thank for the commends i'm creating a fact module instead

remixtj committed:
code linting

remixtj committed:
docs fix

jamescassell commented on lib/ansible/modules/remote_management/ipmi/ipmi_facts.py:
```diff
@@ -0,0 +1,143 @@
+        argument_spec=dict(
+            name=dict(required=True),
+            port=dict(default=623, type='int'),
+            user=dict(required=True, no_log=True),
```
Seems fine to log the user.

jamescassell commented on lib/ansible/modules/remote_management/ipmi/ipmi_facts.py:
```diff
@@ -0,0 +1,143 @@
+    type: str
+    sample: default
+bootdev_persistent:
+    description: If True, system firmware will use this device beyond next boot.
```
```suggestion
    description: If True, system firmware will use the specified device beyond next boot.
```

jamescassell commented on lib/ansible/modules/remote_management/ipmi/ipmi_facts.py:
```diff
@@ -0,0 +1,143 @@
+
+RETURN = '''
+bootdev:
+    description: The boot device name which will be used beyond next boot.
```
```suggestion
    description: The boot device name which will be used for the next boot (and beyond if persistent).
```

jamescassell commented on lib/ansible/modules/remote_management/ipmi/ipmi_facts.py:
```diff
@@ -0,0 +1,143 @@
+    type: bool
+    sample: false
+uefimode:
+    description: If True, system firmware will use UEFI boot explicitly beyond next boot.
```
```suggestion
    description: If True, system firmware will use UEFI boot explicitly.
```

Commit:
Added suggestions by jamescassel

Commit:
Changed added version to 2.9

Akasurde commented:
@remixtj Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to [`community.general`](https://github.com/ansible-collections/community.general). Could you please migrate this PR to the above-mentioned repo? 

If you have further questions please stop by IRC or the mailing list:

* IRC: #ansible on irc.freenode.net
* mailing list: https://groups.google.com/forum/#!forum/ansible-project

Thanks,

needs_info

remixtj commented:
Hello,

I've no time to move this in few days.

Luca
